automation/WinFormsAutomationApp/AuthenticationHelper.cs

```diff
@@ -153,17 +145,66 @@ internal static class AuthenticationHelper
 
         public static async Task<string> AcquireTokenUsingDeviceProfile(Dictionary<string, string> input)
         {
-            Task<string> myTask = Task<string>.Factory.StartNew(() =>
+            Dictionary<string, object> res = new Dictionary<string, object>();
+            IWebDriver driver = null;
+            AuthenticationContext ctx = new AuthenticationContext(input["authority"]);
+            try
             {
-                int count = TokenCache.DefaultShared.Count;
-                Dictionary<string, string> output = new Dictionary<string, string>();
-                output.Add("item_count", count.ToString());
-                var list = TokenCache.DefaultShared.ReadItems().ToJson();
-                output.Add("AccessToken", list);
-                return output.ToJson();
-            });
+                //Get Device Code
+                DeviceCodeResult res1 = await ctx.AcquireDeviceCodeAsync(input["resource"], input["client_id"]);
 
-            return await myTask.ConfigureAwait(false);
+                //Open Chrome Driver in incognito mode
+                ChromeOptions options = new ChromeOptions();
```

Comment:
@TanujaKulal, that doesn't sound like a good reason to have all this code in the sample app itself?
If this code is working in the sample app then it should also work if it was in the test code.
What issues were you hitting when switching contexts? Perhaps we can help fix the issue in the test framework?